alchemist/alchemist-fxui/src/main/kotlin/it/unibo/alchemist/boundary/monitors/AbstractFXDisplay.kt

```diff
@@ -0,0 +1,284 @@
+/**
+ * Main constructor. It lets the developer specify the number of steps.
+ *
+ * @param steps the number of steps
+ * @see .setStep
+ */
+@JvmOverloads constructor(steps: Int = DEFAULT_NUMBER_OF_STEPS) : Pane(), FXOutputMonitor<T, P> {
+
+    private val effectStack: ObservableList<EffectGroup<P>> = FXCollections.observableArrayList()
+    private val mutex = Semaphore(1)
+    private val mayRender = AtomicBoolean(true)
+    private var step: Int = 0
+    @Volatile private var firstTime: Boolean = false
+    private var realTime: Boolean = false
+    @Volatile private var commandQueue: ConcurrentLinkedQueue<() -> Unit> = ConcurrentLinkedQueue()
+    private var viewStatus = DEFAULT_VIEW_STATUS
+    private lateinit var wormhole: BidimensionalWormhole<P>
+    private lateinit var zoomManager: ZoomManager
+    private val interactions: InteractionManager<T, P> by lazy {
+        InteractionManager(
+            this
+        )
+    }
+    private val effectsCanvas = Canvas()
+    /**
+     * Group dedicated for painting the background.
+     */
+    val background = Group()
+
+    init {
+        firstTime = true
+        setStep(steps)
+        val repaintOnResize = ChangeListener<Number> { _, _, _ ->
+            repaint()
+        }
+        widthProperty().addListener(repaintOnResize)
+        heightProperty().addListener(repaintOnResize)
+        effectsCanvas.isMouseTransparent = true
+        effectsCanvas.widthProperty().bind(widthProperty())
+        effectsCanvas.heightProperty().bind(heightProperty())
+        children.addAll(background, effectsCanvas, interactions.canvases)
+    }
+
+    override fun getViewStatus(): FXOutputMonitor.ViewStatus {
+        return this.viewStatus
+    }
+
+    override fun setViewStatus(viewStatus: FXOutputMonitor.ViewStatus) {
+        this.viewStatus = viewStatus
+    }
+
+    override fun getStep(): Int {
+        return this.step
+    }
+
+    /**
+     * {@inheritDoc}.
+     *
+     * @throws IllegalArgumentException if the step is not bigger than 0
+     */
+    @Throws(IllegalArgumentException::class)
+    override fun setStep(step: Int) {
+        if (step <= 0) {
+            throw IllegalArgumentException("The parameter must be a positive integer")
+        }
+        this.step = step
+    }
+
+    override fun isRealTime(): Boolean {
+        return this.realTime
+    }
+
+    override fun setRealTime(realTime: Boolean) {
+        this.realTime = realTime
+    }
+
+    override fun repaint() {
+        mutex.acquireUninterruptibly()
+        try {
+            if (mayRender.get() && isVisible && !isDisabled) {
+                mayRender.set(false)
+                Platform.runLater {
+                    commandQueue.forEach { it() }
+                    mayRender.set(true)
+                }
+            }
+            interactions.repaint()
+        } catch (e: UninitializedPropertyAccessException) {
+            // wormhole hasn't been initialized
```

Comment:
This is a silenced exception and must be fixed. When does it happen? Why?

Reply:
As far as I could tell, this exception doesn't occur. Removed the try-catch.